sonar-db/src/main/java/org/sonar/db/profiling/ProfilingPreparedStatementHandler.java

```diff
@@ -19,43 +19,44 @@
  */
 package org.sonar.db.profiling;
 
-import com.google.common.collect.Lists;
 import java.lang.reflect.InvocationHandler;
 import java.lang.reflect.Method;
 import java.sql.PreparedStatement;
-import java.util.List;
 import org.apache.commons.lang.StringUtils;
 import org.sonar.api.utils.log.Profiler;
 
+import static org.sonar.db.profiling.SqlLogFormatter.FORMATTER;
+
 class ProfilingPreparedStatementHandler implements InvocationHandler {
 
   private final PreparedStatement statement;
-  private final List<Object> arguments;
   private final String sql;
+  private final Object[] sqlParams;
 
   ProfilingPreparedStatementHandler(PreparedStatement statement, String sql) {
     this.statement = statement;
     this.sql = sql;
-    this.arguments = Lists.newArrayList();
-    for (int argCount = 0; argCount < StringUtils.countMatches(sql, "?"); argCount++) {
-      arguments.add("!");
-    }
+    int argCount = StringUtils.countMatches(sql, "?");
+    sqlParams = new Object[argCount];
   }
 
   @Override
   public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
     if (method.getName().startsWith("execute")) {
-      Profiler profiler = Profiler.create(ProfiledDataSource.SQL_LOGGER).start();
+      Profiler profiler = Profiler.createIfTrace(ProfiledDataSource.SQL_LOGGER).start();
```

Comment:
No need. Package `org.sonar.db.profiling` is enabled only when TRACE level is enabled.